core/blockchain.go

```diff
@@ -1769,6 +1782,7 @@ func (bc *BlockChain) insertChain(chain types.Blocks, verifySeals bool) (int, er
 			parent = bc.GetHeader(block.ParentHash(), block.NumberU64()-1)
 		}
 		statedb, err := state.New(parent.Root, bc.stateCache, bc.snaps)
+		statedb.UsePrefetcher(bc.triePrefetcher)
```

Comment:
Can we pass the `prefetcher` pointer to the statedb directly? 

And also I think the `triePrefetcher` only make sense if we use snapshot. If snapshot is disabled, then we should disable this too.

Reply:
The `bc.triePrefetcher` will be nil if it's disabled. The rest of the snapshot-stuff is handled via `bc`, so it's a bit cumbersome to make the connection directly to statedb.